### PR TITLE
fix: add the proxy-protocol option

### LINKED ISSUE #10492: Support proxy_protocol
### Description

Allow the listener to be configured to accept the HAProxy [proxy_protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/), which prefixes each TCP stream with information about the source IP address.

### Discussion

_No response_

### Motivation

The recommended deployment for Keycloak is behind a load-balancer.

The source IP address which Keycloak sees is the IP address of the load-balancer. The standard solution is for the proxy to add an `X-Forwarded-For:` header giving the original source address.  However this means that the load-balancer must be HTTP-aware, *and* it must decrypt HTTPS before proxying, possibly then re-encrypting to a different hostname.

Supporting proxy_protocol would allow layer 4 load balancers such as haproxy or sniproxy to be used.  They can forward TCP connections unmodified, and hence do not need to decrypt TLS or have any certificates of their own.  If necessary, they can inspect SNI (Server Name Indication) information in the stream to select the backend to send to - although in the case of Keycloak with a dedicated proxy and outside IP address, this would not be necessary.

Overall this would be an improvement in security and performance.

### Details

This would require one or more new listener options.

It would be helpful in some scenarios to have both plain and proxy_protocol listeners.  Therefore, maybe something like this:

```
http-enabled=true
http-host=0.0.0.0
http-port=8080
http-proxy-protocol-enabled=true
http-proxy-protocol-host=0.0.0.0
http-proxy-protocol-port=8081
```

### PR DESCRIPTION
**Pull request by shawkins**
closes: #10492

cc @keycloak/cloud-native-maintainers 

Does an integration test for this make sense? The quarkus pr did not include a high level test either.

The quarkus name is use-proxy-protocol, but we generally want our proxy options to all start with proxy, so for us it's just proxy-protocol - but that does read oddly as a boolean. However proxy-use-proxy-protocol seems too long.

We potentially could be opinionated and set this to true when https is enabled and proxy-headers has no value - but we'd to understand if there are any risks / performance overhead before doing that.

@vmuzikar do you want to consider this for v26?

### CODE REVIEW

**Comment by shawkins** [comment]
@vmuzikar updated to proxy-protocol-enabled

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ProxyPropertyMappers.java [hunk 1]
```diff
@@ -21,6 +21,10 @@ public static PropertyMapper<?>[] getProxyPropertyMappers() {
                         .transformer((v, c) -> proxyEnabled(null, v, c))
                         .paramLabel("headers")
                         .build(),
+                fromOption(ProxyOptions.PROXY_PROTOCOL_ENABLED)
+                        .to("quarkus.http.proxy.use-proxy-protocol")
+                        .paramLabel("PROXY protocol")
```
Comment (vmuzikar) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ProxyPropertyMappers.java:26]: Sorry, just one last thing. For consistency, we should have here something like `true|false` like [here](https://github.com/keycloak/keycloak/blob/db4a555ebfac6cbad2c133a83508204194294b9d/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HealthPropertyMappers.java#L16) or [here](https://github.com/keycloak/keycloak/blob/db4a555ebfac6cbad2c133a83508204194294b9d/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/MetricsPropertyMappers.java#L19), or no label at all.

Comment (shawkins) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ProxyPropertyMappers.java]: Corrected now.

**Review by vmuzikar — DISMISSED** [review]
LGTM, thank you!

File: docs/guides/server/reverseproxy.adoc [hunk 2]
```diff
@@ -134,6 +134,16 @@ For example:
 
 <@kc.start parameters="--proxy-headers forwarded --proxy-trusted-addresses=192.168.0.32,127.0.0.0/8"/> 
 
+== PR"OXY Protocol
```
Comment (mabartos) [inline docs/guides/server/reverseproxy.adoc:137]:
```suggestion
== PROXY Protocol
```
I guess it's a typo

**Review by mabartos — COMMENTED** [review]
@shawkins Just a little thing

**Review by mabartos — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**